src/main/java/com/mongodb/kafka/connect/source/heartbeat/HeartbeatManager.java

```diff
@@ -32,31 +41,42 @@
 
 import com.mongodb.client.MongoChangeStreamCursor;
 
-public class HeartbeatManager {
+public class HeartbeatManager implements AutoCloseable {
```

Comment:
This is out of scope for this review regarding the automatic consuming of heartbeat messages.  I'll answer inline - but any further questions lets arrange a zoom call as it will be quicker.  That said, _please file any docs ticket asking for improvements in the documentation that you feel are needed_.  I've seen that you already filed on regarding a duplicated config :+1: 

First some background: The scenario where resumeTokens weren't available after a connector restart was when watching a rarely updated collection on a busy cluster.  In that scenario the resumeToken for the last item updated in the collection can easily roll out of the oplog.  To combat that scenario the postBatchResumeToken functionality was added to changestreams.  The heartbeat mechanism made use of those tokens and publishes them a specific topic for manual consumption. This ticket and change removes the need for manual consumption. KAFKA-176 provides an example using a cli tool on how to do that and its not very user friendly.

Offset partitions naming should not be confused with topics or topic partitions - they are different things.   The Source connector uses the same offset partition (by default based off the connection string) for all SourceRecords. This is required as it always runs off a single task, which uses a single change stream cursor. The heartbeat mechanism also uses that offset partition map when publishing its source records. If it didn't use the same offset partition, then it wouldn't be useful, as when restarting the connector the source task  wouldn't be able to get the latest offset published by the heartbeat manager

The topic the source connector publishes to is derived from the change stream document's namespace. There are no namespaces tied to a postBatchResumeToken, so for heartbeats we use `heartbeat.topic.name` to determine where to publish to.

I hope that clarifies how the source connector uses the offset mechanism and why it only uses a single offset.

I've added DOCSP-15096 to fix the documentation as the sink connector doesn't have a heartbeat configuration and it must have been added in error.

Reply:
Thank you @rozza for the detailed answers here and for the Zoom conversation we had. The only question left is whether or not source records sent to a Kafka topic need to be consumed in order for the offset they bear to be stored by Kafka connect and be available via [`org.apache.kafka.connect.storage.OffsetStorageReader.offset`](https://kafka.apache.org/27/javadoc/org/apache/kafka/connect/storage/OffsetStorageReader.html#offset-java.util.Map-). If consumption is not needed, then this PR may start relying on [Kafka log compaction](https://kafka.apache.org/documentation/#compaction) to prevent unlimited resource consumption instead of consuming events from `heartbeat.topic.name`.